cql3/statements/describe_statement.cc

```diff
@@ -293,6 +289,17 @@ future<std::vector<description>> table(const data_dictionary::database& db, cons
         }
     }
 
+    if (cdc::is_log_for_some_table(db.real_database(), ks, name)) {
+        if (result.size() != 1) {
+            on_internal_error(dlogger, "CDC log table is not supposed to have secondary indexes or MVs derived from it");
+        }
+
+        // If the table the user wants to describe is a CDC log table, we want to print it as a CQL comment.
+        // This way, the user learns about the internals of the table, but it's implicitly implied that
+        // they should not execute the statement.
+        result[0].create_statement = seastar::format("/*\n{}\n*/", *result[0].create_statement);
+    }
```

Comment:
I think there is a better way to write it. You can change the code after the `// table` comment:

```c++
    // table
    replica::schema_describe_helper describe_helper{db};
    auto base_description = schema->describe(describe_helper, with_internals ? describe_option::STMTS_AND_INTERNALS : describe_option::STMTS);
    if (cdc::is_log_for_some_table(db.real_database(), ks, name)) {
        // Transform the `base_description` accordingly and push it to `results`
    } else {
        result.push_back(std::move(base_description));
    }
```

This way, the code will not rely on the assumption that `result.size() == 1` here for cdc tables (you still rely on it, even if you verify it with an assert).

Reply:
Good idea